packages/dd-trace/src/plugins/log_plugin.js

```diff
@@ -6,7 +6,14 @@ const { storage } = require('../../../datadog-core')
 
 const hasOwn = (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop)
 
-function messageProxy (message, holder) {
+function messageProxy (message, holder, constructorId) {
+  // Only apply special stack handling for Winston to avoid breaking other plugins
+  const isWinston = constructorId === 'winston'
```

Comment:
Could any Winston specific handling be moved to the Winston plugin?

Reply:
We would need to copy the whole messageProxy function which would be alot of duplicate code

Reply:
Why? The `isWinston` variable is only used once on the next line. Can't this be done before calling the function?

Reply:
with Ruben's suggestion it's no longer an issue